### ISSUE TITLE
[Firefox Mobile] option buttons sometimes empty in addon properties

### ISSUE THREAD

**Issue by WalterWW** [uBlock-LLC/uBlock#1018]
Sometimes the buttons display correctly, sometimes the text is missing:

http://i.imgur.com/7iBQTrT.jpg

**Comment by gorhill** [comment]
I didn't notice strings were added to `messages.json`... Can you confirm your language is not English?

**Comment by Snow8** [comment]
Having the same issue too sometimes. Dunno what causes it tbh. My language is set to English.

**Comment by WalterWW** [comment]
Language is set to English.

**Comment by gorhill** [comment]
Ok maybe there is something else, but one thing is for sure, all translations are missing for these strings.

**Comment by my-password-is-password** [comment]
I can reproduce.

1.) Go to `about:addons`
2.) Disable uBlock
3.) Enable uBlock

The 2 buttons are blank like in the screenshot from @WalterWW .  Clicking on them does nothing.

**Comment by Deathamns** [comment]
@my-password-is-password Most probably there is an error which prevents the extension to be initialized. Can you view the console errors in Fennec somehow?

Edit: maybe on this address: `chrome://global/content/console.xul`

**Comment by my-password-is-password** [comment]
@Deathamns Sorry for taking long to replay.  Had to boot into win 7 to be able to see my phone the computer.

Have no idea if these belong to ublock.
![ublockfirefoxmobileconsole](https://cloud.githubusercontent.com/assets/6408935/6656204/af85252a-cade-11e4-9ad4-8db6cf6d6899.png)

**Comment by my-password-is-password** [comment]
Ok this is what a got after clearing the console, disable ublock, and then re-enable ublock.

![ublockfirefoxmobileconsole2](https://cloud.githubusercontent.com/assets/6408935/6656217/353a8836-cadf-11e4-9240-f9c497c3350b.png)

**Comment by Deathamns** [comment]
`tab.hasAttribute` is not available on Fennec (fixed in ac09dfa), however I don't think that prevented the extension from loading.

No other error appears when you _enable_ the extension?

**Comment by my-password-is-password** [comment]
Nope.  Tried it again and its the same 4 lines in the console ( 2nd  screenshoot ).

**Comment by my-password-is-password** [comment]
@Deathamns This happens on desktop Firefox too.

![ublockffdesktop](https://cloud.githubusercontent.com/assets/6408935/6656507/20760eb0-caea-11e4-8834-057273fe3dbf.gif)

**Comment by gorhill** [comment]
When refreshing the page the buttons appear correctly.

**Comment by Deathamns** [comment]
I see. Those buttons are activated/filled only when you navigate to this page, because currently we detect when this page opens..

**Comment by gorhill** [comment]
Fixed with 90295e7bef1b699cc3cad22576e5dea024c7c090.

**Comment by WalterWW** [comment]
Using 0.9.2.0 on firefox mobile, the buttons are still sometimes empty.

To reproduce it seems to be enough to open a few tabs, then close all but one and open the tools-addons menu from the dropdown list.

**Comment by AlexVallat** [comment]
I haven't been able to reproduce this, perhaps you could give more detailed steps? What I did:
- Start Firefox Mobile
- Click the New Tab button 5 times
- Click the Close Tab button 5 times
- Click the Tools/Addons menu item
- Click the uBlock addon entry

**Comment by WalterWW** [comment]
Try opening actual websites with new tabs and not just empty tabs...

http://i.imgur.com/Ice6yY1.jpg

**Comment by AlexVallat** [comment]
Sorry, still makes no difference if I insert a step 3 of navigating to google.com (for example). Perhaps you can pin down a minimal case? Starting with Firefox closed, is there a certain number of tabs required - does it happen with just two (the initial, and one opened)? Does it only happen for certain sites? Does the order matter?

**Comment by WalterWW** [comment]
Yeah, I have to admit that it's not easy to consistently reproduce. But it always happens after opening and closing a few websites in new tabs.

Try this: (see edit first)
1. go to https://github.com/gorhill/uBlock/releases
2. long press on that little i for issues to open the issues page in a new tab
3. switch to that tab and then close it
4. open about:addons from the dropdown and open ublock
5. are the buttons empty? if yes, success! if no, close the ublock tab, go to step 2 and repeat until success

I just did this a few times and it seems to happen after about the 4th iteration.

edit: I just figured that it's probably enough to just repeatedly opening the about:addons tab from dropdown, checking ublock, closing the tab and repeating until the buttons are empry. Seems to reproduce the issue as well, so no reason to bother with all the above steps.
